spf/dnscache.go

```diff
@@ -0,0 +1,26 @@
+package spf
+
+type dnsCache map[string]map[string][]string
```

Comment:
Just some speculation, but if we had the ability to do A and AAAA lookups, we may be able to do some CNAME flattening. Which would of course require some automation or something to keep it updated regularly, but a simple cron job may suffice.